src/caselawclient/types.py

```diff
@@ -1,4 +1,11 @@
-from typing import NamedTuple
+from dataclasses import dataclass, field
+from typing import List, NamedTuple
+
+
+@dataclass
+class DocumentCategory:
+    name: str
+    subcategories: List["DocumentCategory"] = field(default_factory=list)
```

Comment:
I'm happy with the potential for arbitrary depth here, and that UI will need to either handle it or decide where to cut it off.